src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -65,40 +80,112 @@ internal MeterProviderSdk(IEnumerable<string> meterSources, int observationPerio
-                    {
-                    }
-
-                    this.listener.RecordObservableInstruments();
-                }
-            });
+            this.observerTask = Task.Run(async () => await this.ObserverTask(token));
+            this.exporterTask = Task.Run(async () => await this.ExporterTask(token));
         }
 
+        internal int ObservationPeriodMilliseconds { get; } = 1000;
+
+        internal int ExportPeriodMilliseconds { get; } = 1000;
+
+        internal List<MeasurementProcessor> MeasurementProcessors { get; } = new List<MeasurementProcessor>();
+
+        internal List<AggregateProcessor> AggregateProcessors { get; } = new List<AggregateProcessor>();
+
+        internal List<MetricProcessor> MetricProcessors { get; } = new List<MetricProcessor>();
+
+        internal List<MetricProcessor> ExportProcessors { get; } = new List<MetricProcessor>();
+
         internal void MeasurementsCompleted(Instrument instrument, object? state)
         {
             Console.WriteLine($"Instrument {instrument.Meter.Name}:{instrument.Name} completed.");
         }
 
-        internal void MeasurementRecorded<T>(Instrument instrument, T value, ReadOnlySpan<KeyValuePair<string, object?>> attribs, object? state)
+        internal void MeasurementRecorded<T>(Instrument instrument, T value, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state)
+            where T : unmanaged
         {
-            Console.WriteLine($"Instrument {instrument.Meter.Name}:{instrument.Name} recorded {value}.");
+            // Run Pre Aggregator Processors
+
+            var measurmentContext = new MeasurementItem(instrument, new DataPoint<T>(value, tags));
+
+            foreach (var processor in this.MeasurementProcessors)
+            {
+                processor.OnEnd(measurmentContext);
+            }
+
+            // Run Aggregator Processors
+
+            foreach (var processor in this.AggregateProcessors)
+            {
+                processor.OnEnd(measurmentContext);
+            }
         }
 
         protected override void Dispose(bool disposing)
         {
             this.cts.Cancel();
+
             this.observerTask.Wait();
+
+            this.exporterTask.Wait();
+
+            this.Export();
+        }
+
+        private async Task ObserverTask(CancellationToken token)
+        {
+            while (!token.IsCancellationRequested)
+            {
+                try
+                {
+                    await Task.Delay(this.ObservationPeriodMilliseconds, token);
+                }
+                catch (TaskCanceledException)
+                {
+                }
+
+                this.listener.RecordObservableInstruments();
+            }
+        }
+
+        private async Task ExporterTask(CancellationToken token)
+        {
+            while (!token.IsCancellationRequested)
+            {
+                try
+                {
+                    await Task.Delay(this.ExportPeriodMilliseconds, token);
```

Comment:
Is the interval expected to be fixed even the the `Export` call takes longer than the `ExportPeriodMilliseconds`?